src/app/courses-dashboard/store/courses.effect.ts

```diff
@@ -14,11 +16,24 @@ import {
   getCourseByIdSuccess,
   addCourse,
   editCourse,
-  manageCourseSuccess
+  manageCourseSuccess,
+  getAuthors,
+  getAuthorsSuccess
 } from './courses.actions';
 
 @Injectable()
 export class CoursesListEffects {
+  getAuhtors$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(getAuthors),
+      mergeMap(({ searchString }) =>
```

Comment:
Check the difference between `mergeMap`, `flatMap` and `switchMap` and decide what operator better to use here (it is not `mergeMap`).